This looks good to me. The test that failed, looks like it was expecting this query, there may be more though. @millerdev any reason we should fetch a form from riak in this case instead of keeping it around?

```
======================================================================
FAIL: corehq.form_processor.tests.test_form_dbaccessor:FormAccessorTestsSQL.test_get_with_attachments
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/form_processor/tests/test_form_dbaccessor.py", line 95, in test_get_with_attachments
    form.get_attachment_meta('form.xml')
  File "/vendor/lib/python2.7/site-packages/django/test/testcases.py", line 88, in __exit__
    query['sql'] for query in self.captured_queries
AssertionError: 0 queries executed, 1 expected
Captured queries were:
```

--- a/corehq/form_processor/tests/test_form_dbaccessor.py
+++ b/corehq/form_processor/tests/test_form_dbaccessor.py
@@ -91,6 +91,10 @@ class FormAccessorTestsSQL(TestCase):
 
     def test_get_with_attachments(self):
         form = create_form_for_test(DOMAIN)
+        form = FormAccessorSQL.get_form(form.form_id)  # refetch to clear cached attachments
+        with self.assertNumQueries(1, using=db_for_read_write(XFormAttachmentSQL)):
+            form.get_attachment_meta('form.xml')
+
         with self.assertNumQueries(1, using=db_for_read_write(XFormAttachmentSQL)):
             form.get_attachment_meta('form.xml')
 

